server/ServerFSM.cpp

```diff
@@ -281,6 +281,84 @@ void ServerFSM::HandleNonLobbyDisconnection(const Disconnection& d) {
     }
 }
 
+bool ServerFSM::EstablishPlayer(const PlayerConnectionPtr& player_connection,
+                                const std::string& player_name,
+                                Networking::ClientType client_type,
+                                const std::string& client_version_string,
+                                const Networking::AuthRoles& roles)
+{
+    std::list<PlayerConnectionPtr> to_disconnect;
+
+    if (player_connection->IsAuthenticated()) {
+        // drop other connection with same name
+        for (ServerNetworking::const_established_iterator it = m_server.m_networking.established_begin();
```

Comment:
could probably use `auto` instead of writing out the verbose iterator type explicitly

Reply:
Fixed.